contrib/auto_explain/auto_explain.c

```diff
@@ -228,9 +226,13 @@ explain_ExecutorStart(QueryDesc *queryDesc, int eflags)
 
 			queryDesc->instrument_options |= INSTRUMENT_CDB;
 
-			INSTR_TIME_SET_CURRENT(starttime);
-			queryDesc->showstatctx = cdbexplain_showExecStatsBegin(queryDesc,
-																   starttime);
+			if (queryDesc->showstatctx == NULL)
+			{
+				instr_time		starttime;
+				INSTR_TIME_SET_CURRENT(starttime);
```

Comment:
now, `starttime` (`querystarttime`) is incremented only if `queryDesc->showstatctx == NULL` ?

Reply:
> now, `starttime` (`querystarttime`) is incremented only if `queryDesc->showstatctx == NULL` ?

Not NULL value for showstatctx may be set in ExplainOnePlan() before this function call. This check is added to do not overwrite showstatctx if the context already exists. INSTR_TIME_SET_CURRENT is only current time setting in starttime. INSTR_TIME_SET_CURRENT affects starttime only. The starttime variable is used for cdbexplain_showExecStatsBegin() only, so starttime initialization does not make sense when cdbexplain_showExecStatsBegin() is not called.